ApplicationLibCode/FileInterface/RifOpmFlowDeckFile.cpp

```diff
@@ -830,6 +835,132 @@ bool RifOpmFlowDeckFile::addOperaterKeyword( std::string          section,
+            case cvf::StructGridInterface::POS_K:
+                return "Z";
+            case cvf::StructGridInterface::NEG_K:
+                return "Z-";
+            default:
+                return "";
+        }
+    };
+
+    // Create the BCCON keyword using OPM's BCCON parser keyword
+    using B = Opm::ParserKeywords::BCCON;
+
+    Opm::DeckKeyword bcconKw( ( Opm::ParserKeyword( B::keywordName ) ) );
+
+    for ( const auto& borderFace : borderCellFaces )
+    {
+        // Convert from 0-based to 1-based Eclipse indexing
+        int i1 = static_cast<int>( borderFace.ijk[0] ) + 1;
+        int j1 = static_cast<int>( borderFace.ijk[1] ) + 1;
+        int k1 = static_cast<int>( borderFace.ijk[2] ) + 1;
+
+        std::string faceStr = faceTypeToString( borderFace.faceType );
+
+        // Create items for the record using proper BCCON item names
+        std::vector<Opm::DeckItem> recordItems;
+
+        recordItems.push_back( RifOpmDeckTools::item( B::INDEX::itemName, borderFace.boundaryCondition ) );
+        recordItems.push_back( RifOpmDeckTools::item( B::I1::itemName, i1 ) );
+        recordItems.push_back( RifOpmDeckTools::item( B::I2::itemName, i1 ) ); // Same as i1 for single cell
+        recordItems.push_back( RifOpmDeckTools::item( B::J1::itemName, j1 ) );
+        recordItems.push_back( RifOpmDeckTools::item( B::J2::itemName, j1 ) ); // Same as j1 for single cell
+        recordItems.push_back( RifOpmDeckTools::item( B::K1::itemName, k1 ) );
+        recordItems.push_back( RifOpmDeckTools::item( B::K2::itemName, k1 ) ); // Same as k1 for single cell
+        recordItems.push_back( RifOpmDeckTools::item( B::DIRECTION::itemName, faceStr ) );
+
+        bcconKw.addRecord( Opm::DeckRecord{ std::move( recordItems ) } );
+    }
+
+    m_fileDeck->insert( insertPos.value(), bcconKw );
+    return true;
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+bool RifOpmFlowDeckFile::addBcpropKeyword( std::string                                               section,
+                                           const std::vector<RigEclipseResultTools::BorderCellFace>& boundaryConditions,
+                                           const std::vector<Opm::DeckRecord>&                       boundaryConditionProperties )
+{
+    if ( m_fileDeck.get() == nullptr ) return false;
+
+    if ( boundaryConditions.empty() ) return true; // Nothing to add
+
+    // Find insertion point within the section
+    auto insertPos = internal::findSectionInsertionPoint( m_fileDeck, section );
+    if ( !insertPos.has_value() )
+    {
+        return false; // Section not found
+    }
+
+    // Create the BCPROP keyword using OPM's BCPROP parser keyword
+    using B = Opm::ParserKeywords::BCPROP;
+
+    Opm::DeckKeyword bcpropKw( ( Opm::ParserKeyword( B::keywordName ) ) );
```

Comment:
We should use
`Opm::DeckKeyword bcpropKw((Opm::ParserKeywords::BCONPROP()))`
to properly initialize the keyword properties